src/common/fetchClient.ts

```diff
@@ -0,0 +1,95 @@
+export interface FetchClientConfig {
+  baseURL: string;
+  headers: Record<string, string>;
+}
+
+export interface FetchRequestConfig<D = any> {
+  params?: Record<string, string>;
+  headers?: Record<string, string>;
+  body?: D;
+}
+
+export interface FetchResponse<T = any> extends Response {
+  data: T;
+}
+
+export default class FetchClient {
+  config: FetchClientConfig;
+
+  constructor(config: FetchClientConfig) {
+    this.config = config;
+  }
+
+  async request(
+    method: string,
+    path: string,
+    config: FetchRequestConfig = {},
+  ): Promise<FetchResponse> {
+    const url = this.buildUrl(path, config.params);
+
+    const response = await fetch(url, {
+      method,
+      headers: {
+        ...this.config.headers,
+        ...(config.headers ?? {}),
+      },
+      body: config.body ? JSON.stringify(config.body) : undefined,
```

Comment:
I know it probably doesn't really make a difference in our usage of this fetch clicent, but is it right to always assume to JSON.stringfy the body data? Wouldn't it be dependent on the http method and also the type of data we are sending?

Reply:
Looked at the Axios source and you're right. They do some more checks around the header type and checking if the body is an object, form data, etc. I'll look at adding a basic check for the body.

Reply:
I added a `prepareRequestBody` function which checks the headers and input data to format the request.